### PR TITLE
dstate: clean separation of UPS-statused based ALARM and modern alarm_*-driven ALARMs

### LINKED ISSUE #2928: dstate/dummy-ups: ALARM status does not clear after dirty entrance into state
Stumbled upon this with `dummy-ups` and the 2.8.3 release version:

Setting `ups.status` to `OL ALARM` correctly registers the alarm state. (Note: I did not expose `ups.alarm` hence as `N/A`)
`Apr 29 14:44:59 TowerOLD upsmon[3362741]: UPS ups@127.0.0.1: one or more active alarms: [[N/A]]` 
Setting `ups.status` back to `OL` does not clear the alarm state, `upsc` (still) reports `ups.status: ALARM OL`
Setting `ups.status` to `OL TRIM` in an attempt to clear the alarm state, `upsc` now reports `ups.status: ALARM OL TRIM`
`Apr 29 14:48:49 TowerOLD upsmon[3362741]: UPS ups@127.0.0.1: trimming incoming voltage` (nothing else)
Setting `ups.status` back to `OL` in another attempt to clear both states,  `upsc` now reports `ups.status: ALARM OL`
`Apr 29 14:52:34 TowerOLD upsmon[3362741]: UPS ups@127.0.0.1: no longer trimming incoming voltage` (nothing else)

I'm not sure if this is limited to `dummy-ups`, I haven't had time to investigate further so far.

### PR DESCRIPTION
**Pull request by desertwitch**
follow up to #2931, fixes #2928 
implements the ideas of https://github.com/networkupstools/nut/pull/2931#issuecomment-2841705269

this PR amends the various involved parts to clean separation of legacy UPS-status coupled `ALARM`
and modern `alarm_`-driven `ALARM`, which are decoupled from UPS status aside from the token itself.

it removes making assumptions on legacy UPS-status coupled `ALARM` and removes workarounds trying
to port any such legacy `ALARM` statuses into the UPS-status decoupled `alarm_` schema, because this would
mean later having to couple them back to UPS-status again to be able to exit these alarm states. (within `dstate`)

this PR allows for both to co-exist peacefully with the only limitation being that a `ups.alarm` does not get
injected when it is not otherwise published directly and manually by the (legacy) driver. `ups.alarm` published
manually by the (legacy) UPS driver will still raise appropriate (seperate) notifications, be tracked by `upsmon`
and function normally as if they were raised by the modern `alarm_` functions (from the `upsmon` perspective).

the `upsmon` does not care where in `dstate` an `ALARM` token was set, nor if done so by legacy or modern method.

the tests now cover both legacy, mixed and modern scenarios respectively, as well as some more curious edge-cases.
further work is in progress to modernize the few remaining legacy drivers to use the `status_` and `alarm_` functions.
this ongoing work will later be published in separate PRs.

### CODE REVIEW

File: tests/driver_methods_utest.c [hunk 1]
```diff
@@ -149,80 +201,179 @@ int main(int argc, char **argv) {
...
+	printf(" test for ups.status empty after status_set() and before status_commit(): '%s'; got alarm, no duplicates?\n", NUT_STRARG(valueStr));
+
 	status_commit();
 
+	/* #18 */
 	valueStr = dstate_getinfo("ups.status");
-	report_0_means_pass(strcmp(valueStr, "ALARM OL BOOST OB"));
-	printf(" test for ups.status with explicit alarm_set(N/A) and another alarm_set(): '%s'; is ALARM reported?\n", NUT_STRARG(valueStr));
+	report_0_means_pass(strcmp(valueStr, "OB LB"));
+	printf(" test for ups.status with just OB and LB set via status_set() and committed: '%s'; got alarm, no duplicates?\n", NUT_STRARG(valueStr));
```
Comment (desertwitch) [inline tests/driver_methods_utest.c:354]: `printf` expectations better as "got empty, no whitespace?" ?

File: tests/driver_methods_utest.c [hunk 2]
```diff
@@ -149,80 +201,179 @@ int main(int argc, char **argv) {
...
-	status_set("LB");	/* Should be ignored */
 	alarm_commit();
+	status_init();
+	status_set("OB");
+	status_set("LB");	/* LB should be honoured when we do not ignorelb */
+
+	/* #17 */
+	valueStr = dstate_getinfo("ups.status");
+	report_0_means_pass(strcmp(valueStr, "\0"));
+	printf(" test for ups.status empty after status_set() and before status_commit(): '%s'; got alarm, no duplicates?\n", NUT_STRARG(valueStr));
```
Comment (desertwitch) [inline tests/driver_methods_utest.c:347]: `printf` expectations better as "got empty, no whitespace?" ?

File: tests/driver_methods_utest.c (hunk 1 above)
Comment (desertwitch) [inline tests/driver_methods_utest.c:354]: `printf` expectations better as "got OB LB?" ?

File: tests/driver_methods_utest.c [hunk 3]
```diff
@@ -149,80 +201,179 @@ int main(int argc, char **argv) {
...
 
-	valueStr = dstate_getinfo("ups.alarm");
-	report_0_means_pass(strcmp(valueStr, "[N/A] [Test alarm 2]"));
-	printf(" test for ups.alarm  with explicit alarm_set(N/A) and another alarm_set(): '%s'; got both alarms, namesake of not-injected is not overwritten\n", NUT_STRARG(valueStr));
+	status_set("FSD");
+
+	/* #19 */
+	valueStr = dstate_getinfo("ups.status");
+	report_0_means_pass(strcmp(valueStr, "OB LB"));
+	printf(" test for ups.status with next token set via status_set() but not yet committed: '%s'; got alarm, no duplicates?\n", NUT_STRARG(valueStr));
```
Comment (desertwitch) [inline tests/driver_methods_utest.c:361]: `printf` expectations better as "got OB LB, no FSD?" ?

File: tests/driver_methods_utest.c [hunk 4]
```diff
@@ -149,80 +201,179 @@ int main(int argc, char **argv) {
...
+
+	status_commit();
+
+	/* #20 */
+	valueStr = dstate_getinfo("ups.status");
+	report_0_means_pass(strcmp(valueStr, "OB LB FSD"));
+	printf(" test for ups.status with FSD token set and now committed: '%s'; got alarm, no duplicates?\n", NUT_STRARG(valueStr));
```
[the remark this answers is not in the document]
Comment (jimklimov) [reply tests/driver_methods_utest.c:368]: right :)

File: tests/driver_methods_utest.c (hunk 2 above)
Comment (jimklimov) [reply tests/driver_methods_utest.c:347]: creepy copy-pasta! :) thanks
